server/sonar-server/src/test/java/org/sonar/server/measure/index/ProjectMeasuresIndexTest.java

```diff
@@ -1029,6 +1030,52 @@ public void facet_languages_contains_only_projects_authorized_for_user() throws
       entry("xoo", 5L));
   }
 
+  @Test
+  public void facet_tags() {
+    index(
+      newDoc().setTags(newArrayList("finance", "offshore", "java")),
+      newDoc().setTags(newArrayList("finance", "javascript")),
+      newDoc().setTags(newArrayList("marketing", "finance")),
+      newDoc().setTags(newArrayList("marketing", "offshore")),
+      newDoc().setTags(newArrayList("finance", "marketing")),
+      newDoc().setTags(newArrayList("finance")));
+
+    Map<String, Long> result = underTest.search(new ProjectMeasuresQuery(), new SearchOptions().addFacets(FIELD_TAGS)).getFacets().get(FIELD_TAGS);
+
+    assertThat(result).containsOnly(
+      entry("finance", 5L),
+      entry("marketing", 3L),
+      entry("offshore", 2L),
+      entry("java", 1L),
+      entry("javascript", 1L));
+  }
+
+  @Test
+  public void facet_tags_is_sticky() {
+    index(
+      newDoc().setTags(newArrayList("finance")),
+      newDoc().setTags(newArrayList("finance")),
+      newDoc().setTags(newArrayList("cpp")));
+
+    Map<String, Long> result = underTest.search(new ProjectMeasuresQuery().setTags(newHashSet("cpp")), new SearchOptions().addFacets(FIELD_TAGS)).getFacets().get(FIELD_TAGS);
+
+    assertThat(result).containsOnly(
```

Comment:
Oh sorry, I didn't see the filter on tags. 
In fact to check stickyness I like to add another data in the doc and check that : 

1. the filter is not taking into account on the tags facet
2. the filter is taking into account on the other facet 

It makes test clearer to me, but it's a personal opinion. 
It's true that your test is testing the stickyness